_studio/shared/umc/codec/vc1_dec/src/umc_vc1_dec_task_store.cpp

```diff
@@ -118,22 +108,13 @@ namespace UMC
                 m_pSHeap = new VC1TSHeap((uint8_t*)m_pMemoryAllocator->Lock(m_iTSHeapID),heapSize);
             }
 
-            m_pSHeap->s_new(&m_pGuardGet,m_iNumFramesProcessing);
+            m_pGuardGet.resize(m_iNumFramesProcessing);
             for (i = 0; i < m_iNumFramesProcessing; i++)
             {
-                m_pSHeap->s_new(&m_pGuardGet[i]);
-
-                vm_mutex_set_invalid(m_pGuardGet[i]);
-                if (VM_OK != vm_mutex_init(m_pGuardGet[i]))
-                    return false;
+                m_pGuardGet[i].reset(new std::mutex);
```

Comment:
BTW, the loop might be rewritten in func. style:
```
std::generate_n(std::back_inserter(m_pGuardGet), m_iNumFramesProcessing,
    []() { return std::unique_ptr<std::mutex>{ new std::mutex{} }; } //std::make_unique since C++ 14
);
```